src/linear_algebra/free_module/finite/basic.lean

```diff
@@ -68,6 +68,9 @@ begin
   simp only [set.image_univ, finset.coe_univ, finset.coe_image, basis.span_eq],
 end
 
+instance module.finite.pi {ι : Type*} [fintype ι] : module.finite R (ι → R) :=
+module.finite.of_basis (pi.basis_fun R _)
```

Comment:
Hmm, should this in fact be generalised? If you have `[fintype ι]` and `M : ι → Type*` and `[∀ i, module R (M i)]` and `[∀ i, module.finite R (M i)]`, then I guess `Π i, M i` is finite by a slightly more complicated proof.

To help lean along, this should then maybe be specialized to another instance
```
instance module.finite.pi {ι : Type*} [fintype ι] (M : Type*) [add_comm_monoid M] [module R M] [module.finite R M] : module.finite R (ι → M) :=
sorry
```
This second instance should directly imply the one that is currently in the diff.

Reply:
You're right, this has nothing to do with free modules, it should go in `ring_theory.finiteness`. I will do it.